docs/protocol/tokenomics.md

```diff
@@ -0,0 +1,101 @@
+* OLAS can be locked for veOLAS tokens for permissionless access to a service whitelist that
+unlocks code owners’ top-ups (incentives in OLAS that boost the developers' ETH rewards).
+```https://github.com/valory-xyz/autonolas-tokenomics/blob/main/docs/Autonolas_tokenomics_audit.pdf
+* OLAS can be used to acquire (on a third-party DEX) LP-tokens that are required for
+bonding. This will enable protocol-owned liquidity and therefore support the protocol’s long-term growth.
+
+## Incentivizing software developers
+
+Let us briefly present how the staking model for agents and component code incentivizes software developers:
+
+1. Developers stake agents and/or components on-chain by [minting they software packages as NFTs](./mint_packages_nfts.md#mint-a-component).
+2. Service owners use staked agents and components to create autonomous
+services. Then they [mint their service](./mint_packages_nfts.md#mint-a-service) in the Autonolas Protocol.
+3. Services can receive ETH donations through the Autonolas Protocol as a sign of recognition for their effort and utility.
+4. With a share of accrued donations, the protocol will reward the staked agents
+and/or components that facilitated such donations. Rewards are distributed proportionally to the code contribution to the ecosystem. Subject to governance approval, a portion of those donations can be eventually used by the DAO for ecosystem growth.
+5. Moreover, when the service owners own a certain threshold of veOLAS (e.g. they
+have locked a certain amount of OLAS for a certain period of time), they are
+considered whitelisted. The protocol will also distribute a share of the OLAS
+inflation to staked components (resp. agents) referenced in the whitelisted
+services.
+
+## Incentivizing bonders
+
+The Autonolas Tokenomics bonding mechanism allows users with enabled Uniswap V2 LP-pair assets to deposit them via the Autonolas depository smart contract and receive OLAS tokens at a discount relative to the price quoted on the relevant DEX after a vesting time. The interest rate on purchased bonds and the discount factor can be adjusted to incentivize or discourage bond demand. The protocol currently incentivizes bonding when there is a large potential output of code production in the ecosystem and a production function is used to measure potential code production during one epoch to establish the interest rate and discount factor.
+
+## Interacting with the protocol tokenomics
+
+The [Autonolas Tokenomics web app](https://tokenomics.autonolas.network/) provides a front-end for **donors**, **bonders** and **developers** to interact with the tokenomics model of the protocol.
+
+### Donate page
+
+Anyone can make a voluntary donation to a given service. Donations to the protocol are distributed to agent or component owners according to the direct contribution measure of them. Developers can later claim their incentive proportionally to the usefulness of their code.
```

Comment:
"according to the direct contribution measure of them" -> according to their direct contribution measure